### PR TITLE
Refactor Commands to be services

### PR DESCRIPTION
**Pull request by renatomefi**
| Q             | A
| ------------- | ---
| Bug fix?      | no
| New feature?  | yes
| BC breaks?    | no
| Deprecations? | fixes some deprecations
| Tests pass?   | yes
| Documented?   | no
| License       | MIT

- Now all commands can be services
- When using the services in the tests avoids the commands auto registering, removing 14 deprecation warnings from the tests
- Using services allows the users to take over or change services by their desires using compiler passes

### CODE REVIEW

**Comment by renatomefi** [comment]
@mcg-web if you know (again) why the tests are failing would be handy, I'll check it tonight otherwise

File: Tests/Functional/Command/DebugCommandTest.php [hunk 1]
```diff
@@ -22,13 +21,11 @@ class DebugCommandTest extends TestCase
     public function setUp()
     {
         parent::setUp();
-        $client = static::createClient(['test_case' => 'mutation']);
-        $kernel = $client->getKernel();
+        $kernel = static::bootKernel(['test_case' => 'global']);
```
Comment (mcg-web) [inline Tests/Functional/Command/DebugCommandTest.php]: replace `global` by `mutation`

**Comment by renatomefi** [comment]
@mcg-web bull's eye again! Thank you :+1: 

it would take me ages to find out the reason! haha

**Comment by mcg-web** [comment]
thanks again :+1:

**Pull request merged**